p4-16/psa/PSA.mdk

```diff
@@ -1150,16 +1152,32 @@ session.
 
 ```
 PortId_t         egress_port;
+MulticastGroup_t multicast_group;
 ClassOfService_t class_of_service;
 bool             truncate;
 PacketLength_t   packet_length_bytes;  /// only used if truncate is true
 ```
 
 The `egress_port` may be any port that can be used for normal unicast
-packets, i.e. any normal port, `PSA_PORT_CPU`, or `PSA_PORT_RECIRCULATE`.  For
-the latter two values, the cloned packet will be sent to the CPU, or
-recirculated at the end of egress processing, as a normal unicast
-packet would at the end of egress processing.
+packets, i.e. any normal port, `PSA_PORT_CPU`, or
+`PSA_PORT_RECIRCULATE`. For the latter two values, the cloned packet
+will be sent to the CPU, or recirculated at the end of egress
+processing, as a normal unicast packet would at the end of egress
+processing.
+
+The `multicast_group` may be any multicast group configured by the
+control plane in the PRE. Sending a packet to a clone session with a
+multicast group configured would cause the cloned packet to be
+replicated to each `egress_port` in the multicast group. If the
+`multicast_group` specified in a clone session is not configured in
+the PRE by the control plane, the cloned packet will not be
+multicasted. A PSA target may need extra effort to check the validity
+of the multicast group in a clone session. Therefore, sending a packet
+to a clone session with un-configured multicast group may have
+performance impact. A clone session can configure both `egress_port`
+and `multicast_group` simultaneously. In addition, an `egress_port`
+may appear multiple times in the same `multicast_group`. Refer to
+section [#sec-multicast] on the details of multicast replication.
```

Comment:
This PR should also include a small change to the table in Section 6.4 "Initial values of packets processed by egress", the table with the caption: "Initial values for packets processed by egress."

Right now it specifies the value for the "instance" field at the start of egress processing to be:

"From PacketReplicationEngine configuration for NM packets.  0 for all other kinds of packets."

Perhaps that should change to something like the following?

"From PacketReplicationEngine configuration for NM packets, or cloned packets that are multicast via a multicast group.  0 for NU packets, and for cloned packets that are not multicast replicated."